### DISCUSSION TITLE
homebridge-wol on Docker homebridge : missing instructions on how to add ssh to docker to run remote commands

### DISCUSSION THREAD

**Discussion by xavbart** [AlexGustafsson/homebridge-wol#153] [Questions] [answered]
Hello. Running homebridge-wol on my Synology via Docker, I eventually groked that even if I can ssh from my NAS to my Mac, the Docker instance is not enabled to do so as it is missing the relevant /bin/ssh and the ssh keys would have to be added to the dockerized environment. 
cf. that issue here https://github.com/oznu/docker-homebridge/issues/113
(I tried following the explanations provided but being a bit thick, I am stil not able to get into a CLI of my docker env)
I guess it would be handy to add a caveat on that config (using HB inside Docker) and even a set of instructions on how to facilitate the setup for ssh.

**Comment by AlexGustafsson** [comment, marked as answer]
I can't necessarily recommend modifying a running Docker container. The "solution" to this is to create your own image by using `oznu/docker-homebridge` as the base:

```Dockerfile
FROM oznu/docker-homebridge

RUN apk add --no-cache openssh-client
```

You may need to alter it a bit, but the idea is that you build that image and run it instead. 

For reference, here is the Dockefile I use:

```Dockerfile
# Use the latest maintainance LTS version
FROM node:12-buster-slim

# Install dependencies
RUN apt-get update && \
  apt-get install -y python3-pip iputils-ping && \
  rm -rf /var/lib/apt/lists/*

USER node
ENV NPM_CONFIG_PREFIX=/home/node/.npm-global

# Install plugins
RUN npm install --global --only=production homebridge homebridge-wol@beta homebridge-config-ui-x

VOLUME /var/homebridge

ENV PATH=/home/node/.npm-global/bin:$PATH

# Run Homebridge on start
CMD homebridge --user-storage-path /var/homebridge | tee /var/homebridge/log
```

I'll include this in #148, however, as it seems quite common to run Homebridge using Docker.

**Reply by xavbart** [reply]
Thanks ! (for the speedy reply and the detailed instructions)
I'll try and make sense of this for my Synology env (still trying to understand how I can work Docker outside of the Synology UI which leaves very low flexibility) but yes, this is a common situation I guess so pointers to the proper set-up will be helpful to a few people.

**Reply by xavbart** [reply]
Ok thanks to your help and the explanation from @oznu here https://github.com/oznu/docker-homebridge/issues/113 I managed to enable with ssh and the proper ssh commands.
(for instance I do have to add `-o StrictHostKeyChecking=no` to each command, although I did already accept the host in CLI from docker while testing the keys, but it seems it is required each time, as it might not access the specific known_hosts file.)
So my sleep command would look like (I only put the display to sleep though) :
`ssh -i /homebridge/keys/docker_home_key -o StrictHostKeyChecking=no USER@MAC_IP_ADDRESS 'pmset displaysleepnow;'
